### PR TITLE
fix(rln-relay): clear nullifier log only if length is over max epoch gap

### PR DESCRIPTION
**Pull request by rymnc**
# Description

Fixes an edge case where the nullifier log was cleared when maxEpochGap == nullifierLog.len

# Changes

- [x] Adds unit tests for varied epoch sizes
- [x] fixes edge case

### CODE REVIEW

**Comment by rymnc** [comment]
> Thanks for this. Would it be possible to add a unit test to catch this edge case?

added in [ef242e7](https://github.com/waku-org/nwaku/pull/2836/commits/ef242e7814bb380ad3964fe4b6296864cc2b7982), previously we did not test for multiple epoch sizes, this test ensures it for 1,5,10,30, and 60.

File: tests/waku_rln_relay/test_waku_rln_relay.nim [hunk 1]
```diff
@@ -886,3 +886,48 @@ suite "Waku rln relay":
...
+        let testEpoch =  wakuRlnRelay.calcEpoch(epochTime() + float(rlnEpochSizeSec * i))
+        wakuRlnRelay.nullifierLog[testEpoch] = testProofMetadataTable
+        check: wakuRlnRelay.nullifierLog.len().uint == i + 1
+
+      # clearing it now will be a no-op
+      wakuRlnRelay.clearNullifierLog()
+
+      check: wakuRlnRelay.nullifierLog.len().uint == rlnMaxEpochGap
+
+      # append one more now
```
Comment (alrevuelta) [inline tests/waku_rln_relay/test_waku_rln_relay.nim:920]: two things:

wondering if all this repeated code can be avoided by reusing the `for i in 0..<rlnMaxEpochGap:`. unsure if removing the `<` may help? afaik the intention here is to add rlnMaxEpochGap + 1 nullifier and see that `clearNullifierLog` doesn't clean them.

and for completeness, perhaps we should check that `clearNullifierLog` does clean the log when expected? meaning i keep adding and `len`()` keeps being `rlnMaxEpochGap`?

**Review by alrevuelta — APPROVED** [review]

**Review by gabrielmer — APPROVED** [review]
Thank you!
